Code/GraphMol/MarvinParse/testMrvToMol.cpp

```diff
@@ -0,0 +1,787 @@
+  bool expectedResult;
+  unsigned int reactantCount;
+  unsigned int agentCount;
+  unsigned int productCount;
+  unsigned int warnings;
+  unsigned int errors;
+
+  RxnTest(std::string fileNameInit, bool expectedResultInit,
+          int reactantCountInit, int agentCountInit, int productCountInit,
+          int warnInit, int errorInit)
+      : fileName(fileNameInit),
+        expectedResult(expectedResultInit),
+        reactantCount(reactantCountInit),
+        agentCount(agentCountInit),
+        productCount(productCountInit),
+        warnings(warnInit),
+        errors(errorInit){};
+};
+
+class SmilesTest {
+ public:
+  std::string name;
+  std::string smiles;
+  bool expectedResult;
+  unsigned int atomCount;
+  unsigned int bondCount;
+
+  SmilesTest(std::string nameInit, std::string smilesInit,
+             bool expectedResultInit, int atomCountInit, int bondCountInit)
+      : name(nameInit),
+        smiles(smilesInit),
+        expectedResult(expectedResultInit),
+        atomCount(atomCountInit),
+        bondCount(bondCountInit){};
+
+  bool isRxnTest() const { return false; }
+};
+
+void testSmilesToMarvin(const SmilesTest *smilesTest) {
+  BOOST_LOG(rdInfoLog) << "testing smiles to marin " << std::endl;
+  std::string rdbase = getenv("RDBASE");
+  std::string fName =
+      rdbase + "/Code/GraphMol/MarvinParse/test_data/" + smilesTest->name;
+
+  class LocalVars  // protect against mem leak on error
+  {
+   public:
+    RWMol *smilesMol;
+
+    LocalVars(){};
+
+    ~LocalVars() { delete smilesMol; }
+  } localVars;
+
+  try {
+    SmilesParserParams smilesParserParams;
+    smilesParserParams.sanitize = true;
+
+    localVars.smilesMol = SmilesToMol(smilesTest->smiles, smilesParserParams);
+    reapplyMolBlockWedging(*localVars.smilesMol);
+
+    TEST_ASSERT(localVars.smilesMol->getNumAtoms() == smilesTest->atomCount);
+    TEST_ASSERT(localVars.smilesMol->getNumBonds() == smilesTest->bondCount);
+
+    // test round trip back to smiles
+    {
+      std::string expectedMrvName = fName + ".expected.smi";
+
+      SmilesWriteParams ps;
+      ps.canonical = false;
+
+      std::string smilesOut = MolToSmiles(*localVars.smilesMol, ps);
+
+      // code to generate the expected files
+
+      // {
+      //   std::ofstream out;
+      //   out.open(fName + ".NEW.smi");
+      //   out << smilesOut;
+      // }
+      std::stringstream expectedMolStr;
+      std::ifstream in;
+      in.open(expectedMrvName);
+      expectedMolStr << in.rdbuf();
+      std::string expectedStr = expectedMolStr.str();
+
+      TEST_ASSERT(expectedStr == smilesOut);
+    }
+
+    {
+      std::string expectedMrvName = fName + ".expected.sdf";
+      std::string outMolStr = "";
+      try {
+        outMolStr = MolToMolBlock(*localVars.smilesMol, true, 0, true, true);
+      } catch (const RDKit::KekulizeException &e) {
+        outMolStr = "";
+      } catch (...) {
+        throw;  // re-trhow the error if not a kekule error
+      }
```

Comment:
There are tests that expect failures - the one in the example above,   a KekulizeException can be thrown.  There are a couple of tests that do throw this exception.  Not all of the tests for it are excercised with the current list of tests,  but it seems prudent to check for it since future tests could throw it.
I don't think :
} catch (...) {
        throw;  // re-throw the error if not a kekule error
      }
does anything and I will remove these.